packages/bloc_concurrency/lib/src/debounce.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+
+/// {@macro debounce}
+EventTransformer<E> debounce<E>([
+  Duration duration = const Duration(milliseconds: 300),
```

Comment:
I think it'd be nice to have the duration be a required, named argument and to align with the API from package:stream_transform. In fact, we already depend on package:stream_transform so I think we should be able to dramatically simplify the implementation here.

I'm thinking the method signature could look something like:

```dart
EventTransformer<E> debounce<E>({
  required Duration duration,
  bool leading = false,
  bool trailing = true,
}) {...}
```

This would also allow us to remove `debounceFirst` altogether and there'd just be a single `debounce` transformer.

Reply:
I'm also just remembering why I never included `debounce` to begin with -- because it's purely rate limiting the event stream but doesn't indicate how the emitted events are processed (e.g. are they still processed concurrently, sequentially, etc?).